src/Controls/src/Core/Handlers/Items2/iOS/LayoutFactory2.cs

```diff
@@ -339,7 +339,7 @@ public static UICollectionViewLayout CreateCarouselLayout(
 					return;
 				}
 
-				var page = (offset.X + sectionMargin) / env.Container.ContentSize.Width;
+				var page = (offset.X + sectionMargin) / (env.Container.ContentSize.Width - sectionMargin * 2);
```

Comment:
Could have negative page calculations causing crashes? For example, using a big value for PeekAreaInsets (like 250). Not usual scenario, but, could we protect it?